src/Controllers/CreateNewEntryController.php

```diff
@@ -30,12 +30,31 @@ public function __invoke(Request $request, Response $response, array $args)
                 'data' => []
             ];
 
-        $AddNewEntry = $request->getParsedBody()['NewBlogEntry'] ?? null;
+        $BlogPost = new Valitron\Validator($_POST);
+        $BlogPost->rule('required', ['title', 'author', 'date', 'post']);
+        $BlogPost->rule('lengthMin', 'post', 5);
+        $BlogPost->rule('lengthMin', 'title', 1);
+        $BlogPost->rule('lengthMin', 'author', 1);
+        $BlogPost->rule('date', 'date');
 
-        if (!$AddNewEntry)
-        {
+        if($BlogPost->validate()) {
+
+            $Result = $this->BlogModel->CreateNewEntry($BlogPost);
+
+            if ($Result) {
+
+                $ResponseData['success'] = true;
+                $ResponseData['message'] = "Your post has been successfully saved!";
+                $ResponseData['data'] = $Result;

```

Comment:
what is this returning, i assume its simply True unless you are sending the blogPost back, if so ,why?